pandas/core/reshape.py

```diff
@@ -1121,6 +1138,18 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
     1  0  1    0
     2  0  0    1
 
+    >>> pd.get_dummies(s1, fill_value=np.nan)
+         a    b
+    0    1    0
+    1    0    1
+    2  NaN  NaN
+
+    >>> pd.get_dummies(s1, fill_value=np.nan, dummy_na=True)
+         a    b  NaN
+    0    1    0    0
+    1    0    1    0
+    2  NaN  NaN    1
```

Comment:
Yes this is the output. If you want zeros, supply fill_value=0. The point here is that a NaN means we do not know what label it is. Putting a zero in the missing spots would indicate that the example is NOT one of those labels. We of course do not know what label the example is since the label is missing.